common/djangoapps/microsite_configuration/backends/database.py

```diff
@@ -0,0 +1,202 @@
+
+        if not microsite:
+            # if no match, then try to find a 'default' key in Microsites
+            try:
+                microsite = Microsite.objects.get(key='default')
+            except Microsite.DoesNotExist:
+                pass
+
+        if microsite:
+            # if we have a match, then set up the microsite thread local
+            # data
+            self._set_microsite_config_from_obj(microsite.subdomain, domain, microsite)
+
+    def get_all_config(self):
+        """
+        This returns all configuration for all microsites
+        """
+        config = {}
+
+        candidates = Microsite.objects.all()
+        for microsite in candidates:
+            values = microsite.values
+            config[microsite.key] = values
+
+        return config
+
+    def get_value_for_org(self, org, val_name, default=None):
+        """
+        This returns a configuration value for a microsite which has an org_filter that matches
+        what is passed in
+        """
+
+        microsite = MicrositeOrgMapping.get_microsite_for_org(org)
+        if not microsite:
+            return default
+
+        # cdodge: This approach will not leverage any caching, although I think only Studio calls
+        # this
+        config = microsite.values
+        return config.get(val_name, default)
+
+    def get_all_orgs(self):
+        """
+        This returns a set of orgs that are considered within a microsite. This can be used,
+        for example, to do filtering
+        """
+
+        # This should be cacheable (via memcache to keep consistent across a cluster)
+        # I believe this is called on the dashboard and catalog pages, so it'd be good to optimize
+        return set(MicrositeOrgMapping.objects.all().values_list('org', flat=True))
+
+    def _set_microsite_config_from_obj(self, subdomain, domain, microsite_object):
+        """
+        Helper internal method to actually find the microsite configuration
+        """
+        config = microsite_object.values
+        config['subdomain'] = subdomain
+        config['site_domain'] = domain
+        config['microsite_config_key'] = microsite_object.key
+
+        # we take the list of ORGs associated with this microsite from the database mapping
+        # tables. NOTE, for now, we assume one ORG per microsite
+        orgs = microsite_object.get_orgs()
+
+        # we must have at least one ORG defined
+        if not orgs:
+            raise Exception(
+                'Configuration error. Microsite {key} does not have any ORGs mapped to it!'.format(
+                    key=microsite_object.key
+                )
+            )
+
+        # just take the first one for now, we'll have to change the upstream logic to allow
+        # for more than one ORG binding
+        config['course_org_filter'] = orgs[0]
+        self.current_request_configuration.data = config
+
+
+class DatabaseMicrositeTemplateBackend(BaseMicrositeTemplateBackend):
+    """
```

Comment:
Here we should include some information about where/how to enable this backend vs. the base.